Lib/dis.py

```diff
@@ -317,19 +318,32 @@ def get_instructions(x, *, first_line=None):
                                    co.co_names, co.co_consts,
                                    linestarts, line_offset, co_positions=co.co_positions())
 
-def _get_const_info(const_index, const_list):
+def _get_const_value(op, arg, co_consts):
+    """Helper to get the value of the const in a hasconst op.
+
+       Returns the dereferenced constant if this is possible.
+       Otherwise (if it is a LOAD_CONST and co_consts is not
+       provided) returns the dis.UNKNOWN sentinel.
+    """
+    assert op in hasconst
+
+    argval = UNKNOWN
+    if op == LOAD_CONST:
+        if co_consts is not None:
+            argval = co_consts[arg]
+    return argval
```

Comment:
I have a patch that exposes the get_const_value function from compile.c in the _opcode module, but I'm on the fence whether it's worth the complication it adds. Let me know if you'd like to see it.

Reply:
Why would you do that? Surely it's easier to replicate its functionality in pure Python, even if we add some new ways to represent constants. It doesn't even take a code object currently. And TBH, should e.g. LOAD_NONE even be counted as having a constant? I'd expect HAS_CONST(LOAD_NONE) might well be False.

Reply:
I think LOAD_NONE should be HAS_CONST, because the peephole optimiser treats it the same as a LOAD_CONST, for things like folding const tuples and conditional jumps that depend on const.

So far I don't think opcode.hasconst is used for anything (it contains only LOAD_CONST). So I thought we would make it include all the splits of LOAD_CONST, and then HAS_CONST is a C macro generated from that (it's new, I just added it).

Reply:
Ah, I see. Currently we take (not the exact notation)
```
LOAD_CONST None
LOAD_CONST 42
BUILD_TUPLE 2
```
and transform it to
```
LOAD_CONST (None, 42)
```
adding `(None, 42)` as a new constant to `co_consts`. We want to still do that even when the first instruction is replaced by `LOAD_NONE`. That makes sense!

We still have the issue that we might end up with unused constants in `co_consts` due to this folding. Inada just fixed this for *trailing* unused constants in GH-28109 (bpo-45056) but for orphaned constants in the middle we have no solution (we'd have to renumber all the LOAD_CONST opargs). Fortunately I think it's not worth it, this is rare.

So you've convinced me that LOAD_NONE should be in `opcode.hasconst`, but not that it's worth exposing that C function.

Reply:
> but not that it's worth exposing that C function.

Yeah, I agree it's OTT in this case. We'll just have to implement it twice.